### PR TITLE
refactor the launch config code to split out from Dart base

### PR DESCRIPTION
**Pull request by devoncarew**
- refactor the launch config code to split out from Dart base

A larger PR, but this sets the stage for future work, including creating a second Flutter launch config to support the Bazel/internal use case. This fully separates out the Flutter and Dart launch config code.

@pq

### CODE REVIEW

File: .idea/inspectionProfiles/Project_Default.xml [hunk 1]
```diff
@@ -23,5 +23,17 @@
...
+      <option name="FIELD" value="true" />
+      <option name="METHOD" value="true" />
+      <option name="CLASS" value="true" />
+      <option name="PARAMETER" value="true" />
+      <option name="REPORT_PARAMETER_FOR_PUBLIC_METHODS" value="true" />
+      <option name="ADD_MAINS_TO_ENTRIES" value="true" />
+      <option name="ADD_APPLET_TO_ENTRIES" value="true" />
+      <option name="ADD_SERVLET_TO_ENTRIES" value="true" />
+      <option name="ADD_NONJAVA_TO_ENTRIES" value="true" />
+    </inspection_tool>
```
Comment (pq) [inline .idea/inspectionProfiles/Project_Default.xml]: I'm guessing adding this is intentional?  (Neutral either way.)

File: src/io/flutter/FlutterBundle.properties [hunk 2]
```diff
@@ -48,3 +48,8 @@ update.module.type=Update Module Type
 updating.module.type.requires.project.reload.proceed=Updating module type requires project reload. Proceed?
 
 waiting.for.flutter=Waiting for debug connection...
+
+path.to.dart.file.not.set=path to Dart source not specified
+dart.file.not.found=Dart file not found: {0}
+not.a.dart.file.or.directory=Specified file is not a Dart file or directory: {0}
+work.dir.does.not.exist=Working directory doesn''t exist: {0}
```
Comment (pq) [inline src/io/flutter/FlutterBundle.properties]: > doesn''t

Seems like a typo?

Also, weren't we keeping these sorted alphabetically?

Comment (devoncarew) [inline src/io/flutter/FlutterBundle.properties]: I think this is to escape the quote? This is what it looks like in the dart properties file.

File: .idea/inspectionProfiles/Project_Default.xml (hunk 1 above)
Comment (devoncarew) [inline .idea/inspectionProfiles/Project_Default.xml]: Yup, intentional, to reduce the warnings in some files.

**Pull request merged**